File: 백준/Silver/13305. 주유소/주유소.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').toString().trim().split('\n');
const N = +input[0];
const dist = input[1].split(' ').map(BigInt);
const price = input[2].split(' ').map(BigInt);
let current = 0;
let cost = 0n;

while (current < N - 1) {
    let next = current + 1;
    let move = dist[current];
    
    while (next < N - 1 && price[next] >= price[current]) {
        move += dist[next];
        next++;
    }
    
    cost += price[current] * move;
    current = next;
}

console.log(String(cost));

File: 백준/Silver/13305. 주유소/주유소.py
N = int(input())
D = list(map(int, input().split()))
C = list(map(int, input().split()))
tc = 0
d = 0
ci = 0
ti = 0

while ci < N-1:
    while ci == ti or (ti < N-1 and C[ci] < C[ti]):
        d += D[ti]
        ti += 1

    tc += d * C[ci]
    d = 0
    ci = ti
    
print(tc)